# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxFontEntry.cpp
# Commit: 2be4a1d5ac07
# Full Hash: 2be4a1d5ac0724aab233faaf469c9b8f5e1d2454
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2022-08-17 15:40:28
# Regressor Bug: 1740530
# File Overlap Count: 1
# Description:
#   Bug 1740530 - patch 1 - Move COLR font support functions from gfxUtils into their own class. r=gfx-reviewers,lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D152036
# ==============================================================================

diff -r 1e2092c319eb -r 2be4a1d5ac07 gfx/thebes/gfxFontEntry.cpp
--- a/gfx/thebes/gfxFontEntry.cpp	Wed Aug 17 10:24:52 2022 +0000
+++ b/gfx/thebes/gfxFontEntry.cpp	Wed Aug 17 10:59:33 2022 +0000
@@ -389,7 +389,7 @@
   auto* colr = GetFontTable(TRUETYPE_TAG('C', 'O', 'L', 'R'));
   auto* cpal = colr ? GetFontTable(TRUETYPE_TAG('C', 'P', 'A', 'L')) : nullptr;
 
-  if (colr && cpal && gfxFontUtils::ValidateColorGlyphs(colr, cpal)) {
+  if (colr && cpal && gfx::COLRFonts::ValidateColorGlyphs(colr, cpal)) {
     if (!mCOLR.compareExchange(nullptr, colr)) {
       hb_blob_destroy(colr);
     }
@@ -1043,9 +1043,9 @@
     uint32_t aGlyphId, const mozilla::gfx::DeviceColor& aDefaultColor,
     nsTArray<uint16_t>& aLayerGlyphs,
     nsTArray<mozilla::gfx::DeviceColor>& aLayerColors) {
-  return gfxFontUtils::GetColorGlyphLayers(GetCOLR(), GetCPAL(), aGlyphId,
-                                           aDefaultColor, aLayerGlyphs,
-                                           aLayerColors);
+  return gfx::COLRFonts::GetColorGlyphLayers(GetCOLR(), GetCPAL(), aGlyphId,
+                                             aDefaultColor, aLayerGlyphs,
+                                             aLayerColors);
 }
 
 typedef struct {